### PR TITLE
Rename and document the ServerPlayNetworkHandler#validateVehicleMove method

### LINKED ISSUE #2603: The ServerPlayNetworkHandler#validateVehicleMove method is incorrect
This method returns `true` if the given movement packet values are invalid. It is used in packet handling for `VehicleMoveC2SPacket` and `PlayerMoveC2SPacket` to disconnect players that send invalid packets.

### PR DESCRIPTION
**Pull request by haykam821**
Fixes #2603

### CODE REVIEW

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping [hunk 1]
```diff
@@ -56,7 +56,13 @@ CLASS net/minecraft/class_3244 net/minecraft/server/network/ServerPlayNetworkHan
 		ARG 2 listener
 	METHOD method_14370 executeCommand (Ljava/lang/String;)V
 		ARG 1 input
-	METHOD method_14371 validateVehicleMove (DDDFF)Z
+	METHOD method_14371 validateMovement (DDDFF)Z
```
Comment (liach) [inline mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]: Shouldn't this be `isMovementInvalid`?

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping (hunk 1 above)
Comment (haykam821) [reply mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]: Yes. Fixed in bde1e3b064e5ea6662d4ec92fcd55b23c991e679

**Comment by liach** [comment]
Sorry but my brain was broke just now, should have been `isMovementInvalid` (I was thinking about the toggling false/true but suggested a bad name)

**Comment by haykam821** [comment]
You're right. The documentation was also incorrect, but I've fixed both in 09c7b0cbf8a782fa80dbc49fbadb0234360e7dd8.

**Review by liach — APPROVED** [review]
Sorry for my brainfarts! Wasn't much into Minecraft recently, so my judgements were faulty.

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping [hunk 2]
```diff
@@ -56,7 +56,13 @@ CLASS net/minecraft/class_3244 net/minecraft/server/network/ServerPlayNetworkHan
...
 	METHOD method_14370 executeCommand (Ljava/lang/String;)V
 		ARG 1 input
-	METHOD method_14371 validateVehicleMove (DDDFF)Z
+	METHOD method_14371 isMovementInvalid (DDDFF)Z
+		COMMENT {@return whether this movement is invalid}
+		COMMENT
+		COMMENT @implNote This method is used to determine whether players sending
+		COMMENT sending {@linkplain PlayerMoveC2SPacket player} and
+		COMMENT {@linkplain VehicleMoveC2SPacket vehicle} movement packets to the server
+		COMMENT should be kicked.
```
Comment (enbrain) [inline mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]: Shouldn't this be `@apiNote`? The comment is how this method is used, not how this method is implemented.

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping [hunk 3]
```diff
@@ -56,7 +56,13 @@ CLASS net/minecraft/class_3244 net/minecraft/server/network/ServerPlayNetworkHan
 		ARG 2 listener
 	METHOD method_14370 executeCommand (Ljava/lang/String;)V
 		ARG 1 input
-	METHOD method_14371 validateVehicleMove (DDDFF)Z
+	METHOD method_14371 isMovementInvalid (DDDFF)Z
+		COMMENT {@return whether this movement is invalid}
+		COMMENT
+		COMMENT @implNote This method is used to determine whether players sending
+		COMMENT sending {@linkplain PlayerMoveC2SPacket player} and
```
Comment (enbrain) [inline mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]: `sending` twice in a row looks like a typo.

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping [hunk 4]
```diff
@@ -56,7 +56,13 @@ CLASS net/minecraft/class_3244 net/minecraft/server/network/ServerPlayNetworkHan
 		ARG 2 listener
 	METHOD method_14370 executeCommand (Ljava/lang/String;)V
 		ARG 1 input
-	METHOD method_14371 validateVehicleMove (DDDFF)Z
+	METHOD method_14371 isMovementInvalid (DDDFF)Z
+		COMMENT {@return whether this movement is invalid}
+		COMMENT
+		COMMENT @implNote This method is used to determine whether players sending
```
Comment (liach) [inline mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]:
```suggestion
		COMMENT @apiNote This method is used to determine whether players
```

**Review by liach — COMMENTED** [review]
This addresses what enbrain points out. Should be good to go.

File: mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping (hunk 3 above)
Comment (haykam821) [reply mappings/net/minecraft/server/network/ServerPlayNetworkHandler.mapping]: Fixed in 3f862a6868a119dc6053e3f1fb71b3fb7573d166

**Comment by liach** [comment]
still should be apiNote, as you talk about its usage.

**Review by YanisBft — APPROVED** [review]

**Review by liach — APPROVED** [review]
Sure, though `@apiNote` is note to callers of this method, while `@implNote` is about what this method is doing

**Pull request merged**